### PR TITLE
[Typing] Convert bool to bit if the column kind is an integer

### PR DESCRIPTION
**Pull request by Tang8330**
## Context

This PR adds the ability to cast a `boolean` value into a `bit` if the column in the destination is an `integer`.

I also rewrote the tests to cover every case and make it more clear what is being tested.

### CODE REVIEW

File: lib/typing/values/string_test.go [hunk 1]
```diff
@@ -1,289 +1,135 @@
 package values
 
 import (
-	"fmt"
```
Comment (Tang8330) [inline lib/typing/values/string_test.go:4]: Rewrote all the tests

File: lib/typing/values/string_test.go [hunk 2]
```diff
@@ -16,274 +15,118 @@ import (
...
-			colVal: birthDateTime,
-			colKind: columns.Column{
-				KindDetails: dateTimeKind,
-			},
-			expectedString: "2022-09-06T03:19:24.942Z",
-		},
+	{
+		// Integer
+		// Floats first.
+		val, err := ToString(float32(45452.999991), columns.Column{KindDetails: typing.Integer}, nil)
```
Comment (nathan-artie) [inline lib/typing/values/string_test.go:88]: Should we be erroring in this case rather than supporting float -> int since it'll result in data loss?

Comment (Tang8330) [reply lib/typing/values/string_test.go:88]: There is precision loss as we're rounding up. A hard failure here doesn't necessarily make sense either as most customers would just create a float compatible column in the destination.

Comment (Tang8330) [reply lib/typing/values/string_test.go:88]: Another reason for this to happen in a legitimate cause could be that they are inconsistent about passing in the optional schema and JSON marshal made these values into `floats` but they're actually ints.

**Review by nathan-artie — APPROVED** [review]

**Pull request merged**